File: pattern.py
n = int(input("enter the number:"))
Range = (2*n)-1

for row in range(1,Range+1):
  col_in_row = row
  if row > n:
    col_in_row = (2*n) - row
  spac_in_row = n - col_in_row
  for spac in range(spac_in_row):
    print(" ",end="")
  for col in range(col_in_row):
    print("* ",end="")
  print()

File: pattern.cpp
#include <iostream>
using namespace std;

void print1(int);
int fib(int);

int main(){
    print1(9);
    cout << fib(8) << endl;
}
void print1(int n){
    for(int row = 1; row < 2*n; row++){
        if(row > n){
            for(int spc = n; spc < row; spc++){
                cout << " ";
            }
            for(int col = row - (n-1); col <= n; col++){
                cout << col << " ";
            }
        }
        else{
             for(int spc = row; spc < n; spc++){
            cout << " ";
        }
        for(int col = (n-row)+1; col <= n; col++){
            cout << col << " ";
        }
        }
       
        cout << endl;
    }
}

int fib(int n){
   return n < 2 ? n : fib(n-1)+fib(n-2);
}
   

